src/helpers/fetch.js

```diff
@@ -1,34 +1,45 @@
-const fetch = (setOptions, callback = false) => {
-	let defaults = {
+const fetchWrapper = (options, callback = false) => {
+	const defaults = {
 		url: window.location.pathname + window.location.search,
 		method: 'GET',
-		data: null,
 		headers: {}
 	};
 
-	let options = {
+	const { url, method, data, body, headers } = {
 		...defaults,
-		...setOptions
+		...options
 	};
 
-	let request = new XMLHttpRequest();
+	const controller = new AbortController();
+	const signal = controller.signal;
 
-	request.onreadystatechange = function() {
-		if (request.readyState === 4) {
-			if (request.status !== 500) {
-				callback(request);
-			} else {
-				callback(request);
+	const init = { method, headers, body: body || data, signal };
+
+	fetch(url, init)
+		.then((response) => {
+			return response.text().then((html) => ({ response, html }));
+		})
+		.then(({ response, html }) => {
+			// Compatibility layer for other methods expecting XHR properties
+			response.responseURL = response.url || url;
+			response.responseText = html;
+			callback(response);
+		})
+		.catch((error) => {
+			// Handle intentionally aborted requests
+			if (signal.aborted && signal.reason === 'aborted') {
+				callback({ error: 'aborted' });
+				return;
 			}
-		}
+			// Handle all other errors
```

Comment:
I just assumed that `fetch` can error for other reasons as well (e.g. network timeout?) That's why I added the callback here as well.